File: 0476. Number Complement.cpp
class Solution
{
public:
    int findComplement(int num)
    {
        if (num == 0)
        {
            return 1;
        }
        
        bool lz = true;
        
        for (int i = 31; i >= 0; --i)
        {
            if (lz and num >> i & 1)
            {
                lz = false;
            }
            
            if (not lz)
            {
                num ^= 1 << i;
            }
        }
        
        return num;
    }
};

File: 0476. Number Complement.cs
public class Solution
{
    public int FindComplement(int num)
    {
        if (num != 0)
        {
            var ind = 31;

            while ((num >> ind & 1) != 1)
            {
                --ind;
            }

            while (ind >= 0)
            {
                num ^= 1 << ind;
                --ind;
            }
    
            return num;
        }
        else
        {
            return 1;
        }
    }
}